enhancements/hybrid-helm-operator.md

```diff
@@ -0,0 +1,260 @@
+---
+title: enable-hybrid-helm-operators
+authors:
+  - "@fabianvf"
+  - "@varshaprasad96
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-05-10
+last-updated: 2021-05-12
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# enable-hybrid-helm-operators
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+
+1. How do we handle migrations for existing Helm-based Operators? For the first iteration
+   requiring users to manually migrate charts and watches.yaml changes should be enough,
+   but longer term we may want to support a command to automatically migrate this over,
+   perhaps as an initialization flag to `operator-sdk init`, similar to how the `--helm-chart`
+   argument works for the helm plugin.
+1. Where should the helm library code live? Likely it should either go in `operator-frameowkr/operator-lib`,
```

Comment:
If it lived in operator-lib, the two concerns I have are:
1. Releasing/versioning. It is totally possible to tag helm-operator releases with `helm/<version>` but handling a variety of git tags in release pipelines can be messy. Perhaps this problem can be solved with git submodules.
2. Issues for both helm-operator and operator-lib libraries will exist in the same repo, which could stack up.

I am not saying operator-lib is a suboptimal choice; the new helm-operator code is supposed to be a library, so operator-lib is the logical spot. Just throwing out some considerations.

Reply:
I think another positive to having a separate helm-operator repo that the library could live in is that it will provide a natural place for the helm-operator plugin to land once we actually do the repo split. I think after the repo split it would feel kind of strange to be importing highly helm-specific logic from the generic operator-lib.